### PR TITLE
Reindex from Remote encoding

### PR DESCRIPTION
**Pull request by henningandersen**
Removed the leniency when encoding remote reindex search requests that
was introduced in 7.x. All index-names are now encoded before being sent
to the remote host.

Follow-up to #40303

### CODE REVIEW

File: docs/reference/migration/migrate_8_0/reindex.asciidoc [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[float]
+[[breaking_80_reindex_changes]]
+=== Reindex changes
+
+Reindex from remote would previously allow URL encoded index-names and not re-encode them when generating the search
```
Comment (jasontedor) [inline docs/reference/migration/migrate_8_0/reindex.asciidoc]: This is a drive-by comment only, but we like to wrap our docs at eighty columns. 😄

Comment (henningandersen) [reply docs/reference/migration/migrate_8_0/reindex.asciidoc]: Thanks @jasontedor , fixed in 67dd80f6f487e259454f24a8efc653c60df7ca10.

**Review by Tim-Brooks — APPROVED** [review]
LGTM

**Comment by henningandersen** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample
